src/action.js

```diff
@@ -73,19 +78,37 @@ async function action(payload) {
     (report) => Math.floor(report.total) < minimumCoverage
   );
 
+  const belowThresholdForAnyFile = reports.some((report) => {
+    const allCoveragesInReport = report.files.map((r) => Math.floor(r.total));
+    const smallestCoverage = Math.min(...allCoveragesInReport);
+    return smallestCoverage < minimumCoverage;
+  });
+
   if (pullRequestNumber) {
     await addComment(pullRequestNumber, comment, reportName);
   }
-  await addCheck(
-    comment,
-    reportName,
-    commit,
-    failBelowThreshold ? (belowThreshold ? "failure" : "success") : "neutral"
-  );
+
+  let conclusion = "neutral";
+
+  if (failBelowThreshold) {
+    conclusion = belowThreshold ? "failure" : "success";
+  }
+
+  if (failForIndividualFileBelowThreshold) {
+    conclusion = belowThresholdForAnyFile ? "failure" : "success";
+  }
+
+  await addCheck(comment, reportName, commit, conclusion);
 
   if (failBelowThreshold && belowThreshold) {
     core.setFailed("Minimum coverage requirement was not satisfied");
   }
+
+  if (failForIndividualFileBelowThreshold && belowThresholdForAnyFile) {
+    core.setFailed(
+      "Minimum coverage requirement was not satisfied for one or more files"
+    );
+  }
```

Comment:
This code isn't terribly DRY. I'm happy to refactor it if you like, but it should be pretty clear what I'm changing with this first commit.